### PR TITLE
Introduce Lotus.root

### PR DESCRIPTION
**Pull request by LeoTeam**

### CODE REVIEW

File: test/root_test.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+require 'test_helper'
+require 'lotus/root'
+
+describe 'Lotus.root' do
```
Comment (joneslee85) [inline test/root_test.rb]: I think you should do

```
  describe `Lotus` do
    describe `.root` do
    end
  end
```

File: test/root_test.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+  let(:pathname) {Lotus.root}
+  it 'return root path' do
```
Comment (joneslee85) [inline test/root_test.rb]: s/return/returns

File: test/root_test.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+require 'test_helper'
+require 'lotus/root'
+
+describe 'Lotus.root' do
+  let(:pathname) {Lotus.root}
```
Comment (joneslee85) [inline test/root_test.rb]: please add empty space after `{`

File: lib/lotus/root.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+require 'pathname'
+
+module Lotus
+  def self.root
+    Lotus::Configuration.new.root
```
Comment (AlfonsoUceda) [inline lib/lotus/root.rb]: IMO, we shouldn't create a new instance of configuration, we have to take the current cofiguration

Comment (joneslee85) [inline lib/lotus/root.rb]: I don't the above implementation is correct though. What you did above is to retrieve the root value of an app (aka slice), what we wanted here is the root level of the container ( the directory that `config.ru` reside )

What I'd like to suggest is to look into `Lotus::Loader`, there you can work out that we should introduce a root setter for `configuration` perhaps? Let me know if you need any assistant on this ticket.

Comment (AlfonsoUceda) [inline lib/lotus/root.rb]: Then you have lotus container, https://github.com/lotus/lotus/blob/master/lib/lotus/container.rb inside you have access to configuration

**Comment by joneslee85** [comment]
```
module Lotus
  def self.root
    @root
  end

  def self.root=(path)
    @root = path
  end
end

module Lotus
  class Configuration

    def load!(namespace = nil)
      @namespace = namespace
      evaluate_configurations!

      # set Lotus.root
      if !Lotus.root
        Lotus.root = self.root
      end

      self
    end
  end
end
```

**Comment by jodosha** [comment]
@LeoTeam @mymai91 This approach is wrong :cry:
You use internally `Environment#root`. If you read its API docs it says: 

> It defaults to the current working directory.
   Lotus assumes that all the commands are executed from there.

That means, it delegates to the current directory, **assuming** that all the commands are ran from there.
**The goal of this feature, is to remove that assumption.**

How to manually test it:

```shell
lotus new bookshelf --lotus-head=true
cd bookshelf
```

Edit `Gemfile` with:

```ruby
gem 'lotusrb',                           github: 'LeoTeam/lotus', branch: 'lotus-root'
```

Then

```shell
bundle
cd apps/web
bundle console
```

Test it:

```ruby
irb(main):001:0> Lotus.root
=> #<Pathname:/Users/luca/Desktop/bookshelf/apps/web>
```

It returns `/Users/luca/Desktop/bookshelf/apps/web`, but it should be `/Users/luca/Desktop/bookshelf`.

/cc @joneslee85

**Comment by jodosha** [comment]
@LeoTeam @mymai91 Can you please implement it here? https://github.com/lotus/lotus/blob/master/lib/lotus.rb Thank you!

/cc @joneslee85

**Comment by jodosha** [comment]
@LeoTeam Thank you. Both me and @joneslee85 agree to rely on `Pathname.pwd` for now. Merging this :space_invader:

**Comment by mymai91** [comment]
@jodosha @joneslee85 Good to know. Thanks.

**Pull request merged**